cpp/oneapi/dal/algo/decision_forest/common.hpp

```diff
@@ -459,6 +459,72 @@ class descriptor : public detail::descriptor_base<Task> {
     }
 };
 
+/// Struct containing base description of node in descision tree
+struct node_descriptor {
+    std::int64_t level = 0; /// Number of connections between the node and the root
+    double impurity =
+        0.; /// Measure of the homogeneity of the response variable at the node (i.e., the value of the criterion)
+    std::int64_t node_sample_count = 0; /// Number of samples at the node
+};
+
+/// Struct containing description of split node in descision tree
+struct split_node_descriptor : public node_descriptor {
+    std::int64_t feature_index = 0; /// Feature used for splitting the node
+    double feature_value = 0.; /// Threshold value at the node
+};
+
+template <typename Task>
+struct leaf_node_descriptor;
+
+/// Struct containing description of leaf node in classification descision tree
+template <>
+struct leaf_node_descriptor<task::classification> : public node_descriptor {
+    std::int64_t label = 0; /// Label to be predicted when reaching the leaf
+    const double* prob = nullptr; /// Probabilities estimation for the leaf
+};
+
+/// Struct containing description of leaf node in regression descision tree
+template <>
+struct leaf_node_descriptor<task::regression> : public node_descriptor {
+    double label = 0.; /// Label to be predicted when reaching the leaf
+};
+
+template <typename T>
+struct is_leaf_node_descriptor {
```

Comment:
The same consideration about `descriptor` here. Maybe `is_leaf_node`?

Reply:
used info instead of descriptor